tests/spawner/test_spawner.py

```diff
@@ -1,54 +1,84 @@
-import sys
-import subprocess
-
+import contextlib
 import os
+import pwd
+import shutil
+import subprocess
+import sys
+import time
+import tempfile
 import unittest
 from unittest import mock
-import pwd
 
+from tornado.ioloop import IOLoop
 from jupyterhub import orm
-from remoteappmanager.spawner import Spawner
+
+from remoteappmanager.spawner import Spawner, VirtualUserSpawner
 from tests import fixtures
 
 
+def start_spawner(spawner):
+    io_loop = IOLoop.current()
+    io_loop.run_sync(spawner.start)
```

Comment:
Ok, if it's because of a race condition with the shell spawn, then keep the sleep.